internal/controller/accessrequest_controller.go

```diff
@@ -111,20 +125,32 @@ func (r *AccessRequestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	application, err := r.getApplication(ctx, ar)
 	if err != nil {
+		// TODO send an event to explain why the access request is failing
 		return ctrl.Result{}, fmt.Errorf("error getting Argo CD Application: %w", err)
 	}
 
+	roleTemplate, err := r.getRoleTemplate(ctx, ar)
+	if err != nil {
+		// TODO send an event to explain why the access request is failing
+		return ctrl.Result{}, fmt.Errorf("error getting RoleTemplate %s: %w", ar.Spec.RoleTemplateName, err)
+	}
+
+	renderedRt, err := roleTemplate.Render(application.Spec.Project, application.GetName(), application.GetNamespace(), application.Spec.Destination.Namespace)
```

Comment:
The application destination namespace is not used in policies to my recollect. Only the Application resource namespaces is use for apps-in-any-namespace feature that might be required. 
Maybe add a comment in the code to explain why `application.Spec.Destination.Namespace` is necessary until a markdown doc exist to document it

Reply:
Good point. I though the destination namespace was used in the policies but checking the docs it is indeed only the app namespace. Removing it from the AppSpec..